File: Lighthouse.Backend/Lighthouse.Backend/Models/OptionalFeatures/OptionalFeature.cs
using Lighthouse.Backend.Services.Interfaces;

namespace Lighthouse.Backend.Models.OptionalFeatures
{
    public class OptionalFeature : IEntity
    {
        public required int Id { get; set; }

        public string Key { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public required bool Enabled { get; set; }

        public bool IsPreview { get; set; } = false;
    }
}

File: Lighthouse.Frontend/src/models/OptionalFeatures/OptionalFeature.ts
import { z } from "zod";

export interface IOptionalFeature {
	name: string;
	key: string;
	id: number;
	description: string;
	enabled: boolean;
	isPremium: boolean;
	isPreview: boolean;
}

export const OptionalFeatureSchema = z.object({
	name: z.string(),
	key: z.string(),
	id: z.number(),
	description: z.string(),
	enabled: z.boolean(),
	isPremium: z.boolean(),
	isPreview: z.boolean(),
});
